rust/src/transformations/polars/make_select_trans/mod.rs

```diff
@@ -0,0 +1,273 @@
+use polars::prelude::*;
+
+use crate::core::{Function, StabilityMap, Transformation};
+use crate::domains::{Context, ExprDomain, LazyFrameDomain};
+use crate::error::*;
+use crate::metrics::{IntDistance, SymmetricDistance};
+
+/// Make a Transformation that applies list of transformations to a Lazy Frame.
+///
+pub fn make_select_trans(
+    input_domain: LazyFrameDomain,
+    input_metric: SymmetricDistance,
+    transformations: Vec<
+        Transformation<ExprDomain, ExprDomain, SymmetricDistance, SymmetricDistance>,
+    >,
+) -> Fallible<Transformation<LazyFrameDomain, LazyFrameDomain, SymmetricDistance, SymmetricDistance>> where
+{
+    let fist_transformation = transformations
+        .first()
+        .ok_or_else(|| err!(MakeTransformation, "transformation list cannot be empty"))?;
+
+    let output_domain = fist_transformation.output_domain.lazy_frame_domain.clone();
+
+    let output_metric = fist_transformation.output_metric.clone();
+
+    if transformations
+        .iter()
+        .any(|t| t.input_domain.lazy_frame_domain != input_domain)
+    {
+        return fallible!(MakeTransformation, "input domains do not match");
+    }
```

Comment:
I think it would be better to define a `let expr_input_domain = ExprDomain { ..., context: Context::Select, selected_column: None };` above, and then check that `t.input_domain` matches `expr_input_domain`. This ensures that the input context and selected column are all correct.